graphql-mode.el

```diff
@@ -388,6 +389,59 @@ This is the function to be used for the hook `completion-at-point-functions'."
      (1 font-lock-variable-name-face)))
   "Font Lock keywords.")
 
+;;; Edit headers functionality:
+
+(defun graphql-edit-headers ()
+  "Edit graphql request headers interactively in a dedicated buffer.
+
+Open a buffer to edit `graphql-extra-headers'.  The contents of this
+buffer take precedence over the setting in `graphql-extra-headers'
+when sending a request."
+  (interactive)
+  (unless (equal major-mode 'graphql-mode)
+    (error "Not in graphql-mode, cannot edit headers"))
+  (let ((extra-headers-buffer
+         (concat "*Graphql Headers for " (buffer-name) "*")))
+    (pop-to-buffer extra-headers-buffer)
+    (if (and (string-empty-p (buffer-string)) graphql-extra-headers)
+        (progn
+          (insert (json-serialize graphql-extra-headers))
```

Comment:
I think there is a problem here.

We should make the `graphql-extra-headers` buffer-local variable.  Now it works because it's global. But If I use this file for instance:

```graphql
{
  search(query: "project:kubernetes/9", type: ISSUE, first: 1, after: null) {
    issueCount
  }
}
# Local Variables:
# graphql-extra-headers: (("Authorization" . "bearer f4489a81b6ee64a7f7da1cbc9526525b50575d54"))
# graphql-url: "https://api.github.com/graphql"
# graphql-variables-file: "graphql-variables"
# End:
```

the editor buffer won't get the local value.  Maybe it is also because `pop-to-buffer` changes the buffer.

Reply:
Yes, that's true. Do you think we should change the declaration of `graphql-extra-headers` to `defvar-local`?

Reply:
I don't personally use the customize interface, but it's nice to keep `defcustom`.  I think calling `make-variable-buffer-local` should work as well?

If you want to try to make it work like this it would be great. But let me know if you want me to help.

Reply:
I ended up using `setq-local`, as that calls `make-variable-buffer-local` under the hood anyway. We now have to remember the parent buffer, so I did end up introducing a local variable for that, as I'm not sure how else to get at it (and I saw several other packages did it that way, too.)